contracts/external/cw-fund-distributor/src/contract.rs

```diff
@@ -323,93 +310,88 @@ pub fn execute_claim_all(deps: DepsMut, env: Env, sender: Addr) -> Result<Respon
     if !funding_deadline.is_expired(&env.block) {
         return Err(ContractError::ClaimDuringFundingPeriod {});
     }
-    let relative_share = get_relative_share(deps.as_ref().borrow(), sender.clone());
-
-    let cw20s: Vec<(Addr, Uint128)> = CW20_BALANCES
-        .range(deps.storage, None, None, Order::Descending)
-        .map(|cw20| cw20.unwrap())
-        .collect();
+    let relative_share = get_relative_share(&deps.as_ref(), sender.clone())?;
 
-    let natives: Vec<(String, Uint128)> = NATIVE_BALANCES
-        .range(deps.storage, None, None, Order::Descending)
-        .map(|native| native.unwrap())
+    let cw20s: Vec<Result<(Addr, Uint128), StdError>> = CW20_BALANCES
```

Comment:
how would you feel about for this method, after we collect the list of cw20s to claim, just calling `ExecuteMsg::ClaimCw20s` and `ExecuteMsg::ClaimNatives` on the contract itself? that might reduce the amount of code duplication here.

1. collect cw20s and natives to claim.
2. call claim methods on self.

Reply:
I tried something close to that, lmk